test/integration/auth/svcaccttoken_test.go

```diff
@@ -567,6 +584,99 @@ func TestServiceAccountTokenCreate(t *testing.T) {
 
 		doTokenReview(t, cs, treq, true)
 	})
+
+	t.Run("a token is valid against the HTTP-provided service account issuer metadata", func(t *testing.T) {
+		var token string
+
+		sa, del := createDeleteSvcAcct(t, cs, sa)
+		defer del()
+
+		t.Run("get token", func(t *testing.T) {
+			treq := &authenticationv1.TokenRequest{
+				Spec: authenticationv1.TokenRequestSpec{
+					Audiences: []string{"api"},
+				},
+			}
+
+			resp, err := cs.CoreV1().ServiceAccounts(sa.Namespace).CreateToken(sa.Name, treq)
+			if err != nil {
+				t.Fatalf("unexpected err creating token: %#v", resp)
+			}
+			token = resp.Status.Token
+		})
+		if token == "" {
+			t.Fatal("no token")
+		}
+
+		metadata := struct {
+			Issuer string `json:"issuer"`
+			Jwks   string `json:"jwks_uri"`
+		}{}
+		t.Run("get metadata", func(t *testing.T) {
+			resp := rc.Get().AbsPath("/.well-known/openid-configuration").Do()
+			if err := resp.Error(); err != nil {
+				t.Fatal(err)
+			}
+			b, err := resp.Raw()
+			if err != nil {
+				t.Fatal(err)
+			}
+			md := bytes.NewBuffer(b)
+			t.Logf("raw metadata response:\n---%s\n---", md.String())
+			if md.Len() == 0 {
+				t.Fatal("empty response for metadata")
+			}
+			if err := json.NewDecoder(md).Decode(&metadata); err != nil {
+				t.Fatal("could not decode metadata: ", err)
+			}
+			if len(metadata.Issuer) == 0 || len(metadata.Jwks) == 0 {
+				t.Fatal("invalid metadata")
+			}
+		})
+
+		var key interface{}
+		t.Run("get jwks", func(t *testing.T) {
+			req := rc.Get().RequestURI(metadata.Jwks)
+			t.Log("fetching from", req.URL())
+			resp := req.Do()
+			if err := resp.Error(); err != nil {
+				t.Fatal(err)
+			}
+			b, err := resp.Raw()
+			if err != nil {
+				t.Fatal(err)
+			}
+			ks := bytes.NewBuffer(b)
+			if ks.Len() == 0 {
+				t.Fatal("empty jwks")
+			}
+			t.Logf("raw JWKS: \n---\n%s\n---", ks.String())
+			jwks := jose.JSONWebKeySet{}
+			if err := json.NewDecoder(ks).Decode(&jwks); err != nil {
+				t.Fatal("could not decode JWKS: ", err)
+			}
+			if len(jwks.Keys) != 1 {
+				t.Fatalf("len(jwks.Keys) = %d, want 1", len(jwks.Keys))
+			}
+			key = jwks.Keys[0]
+		})
+
+		tok, err := jwt.ParseSigned(token)
+		if err != nil {
+			t.Fatalf("could not parse token %q: %v", token, err)
+		}
+		var claims jwt.Claims
+		if err := tok.Claims(key, &claims); err != nil {
```

Comment:
key could be nil here. I'd suggest getting rid of the subsubtests. I'm not sure that they are actually helpful.